[PyTorch] Make tls_local_dispatch_key_set inlineable (reapply)

swolchok opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#49412 [PyTorch] Make tls_local_dispatch_key_set inlineable (reapply)**
* #49371 [PyTorch] Use .sizes() isntead of .size() in cat_serial_kernel_impl
* #49368 [PyTorch] Use .sizes() instead of .size() in _cat_out_cpu
* #49364 [PyTorch] Avoid extra Tensor refcounting in _cat_out_cpu
* #49359 [PyTorch] Merge CoinflipTLS into RecordFunctionTLS
* #49355 [PyTorch] Avoid move-constructing a List in listConstruct

FLAGS_disable_variable_dispatch had to go, but it looks like the only user was some benchmarks anyway.

Differential Revision: [D25547962](https://our.internmc.facebook.com/intern/diff/D25547962/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25547962/)!

swolchok committed:
[PyTorch] Make tls_local_dispatch_key_set inlineable (reapply)

FLAGS_disable_variable_dispatch had to go, but it looks like the only user was some benchmarks anyway.

Differential Revision: [D25547962](https://our.internmc.facebook.com/intern/diff/D25547962/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25547962/)!

[ghstack-poisoned]

ezyang commented:
Yeah.... Scott, I'm pretty sure this error is why it wasn't inlined on Windows. Maybe that just means we should only inline on Linux though.

swolchok committed:
Don't inline on Windows on "[PyTorch] Make tls_local_dispatch_key_set inlineable (reapply)"


FLAGS_disable_variable_dispatch had to go, but it looks like the only user was some benchmarks anyway.

Differential Revision: [D25547962](https://our.internmc.facebook.com/intern/diff/D25547962/)

**NOTE FOR REVIEWERS**: This PR has internal Facebook specific changes or comments, please review them on [Phabricator](https://our.internmc.facebook.com/intern/diff/D25547962/)!

[ghstack-poisoned]

ezyang reviewed (approved):
okey dokey
